dao/pom.xml

```diff
@@ -52,6 +52,17 @@
             <artifactId>jackson-datatype-jsr310</artifactId>
             <version>${jackson.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.4.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.scalatest</groupId>
+            <artifactId>scalatest_${scala.compat.version}</artifactId>
+            <version>${scalatest.version}</version>
+            <scope>provided</scope>
```

Comment:
Shouldn't `scalatest` be scoped `test`?

Reply:
Also, don't you inherit this for free from [`parent-pom`](https://github.com/AbsaOSS/enceladus/blob/develop/pom.xml#L127)?

Reply:
Indeed. We don't need that dependency in DAO. Will remove it.

But curiously enough such exact 'provided' dependency is present in Utils. That's because of `SparkJobRunnerMethods` - a cute reflection/loader trick to run Spark jobs as tests so that provided dependencies are also loaded.

We don't use `SparkJobRunnerMethods` in DCE, but it is a beautiful piece of code that I cannot fully comprehend, but just use and enjoy.

Now here is a dilemma. We can
1. Leave `SparkJobRunnerMethods` in place so we know which project/module to search for when we need it. And let contributors marvel our god-like smartness. Utils is a collection of generic tools after all
2. Remove `SparkJobRunnerMethods` and scala-test dependency - the only non-inherited dependency of Utils.

I'm slightly more inclined towards 1.

Reply:
Agreed, let's keep `SparkJobRunnerMethods`!